cmd/juju/model/defaultscommand.go

```diff
@@ -221,17 +556,28 @@ func (c *defaultsCommand) verifyKnownKeys(client defaultsCommandAPI) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	keys := func() []string {
-		if c.keys != nil {
-			return c.keys
-		}
+	// allKeys gets the all the set and reset keys to validate. This should get
+	// caught in Init so maybe we don't need it here too.
+	// TODO(reviewer) Should we keep this and the init check?
```

Comment:
it's still necessary to check that they're known, but not that they're duplicated. also, the functional literal seems unnecessary? wouldn't the following suffice?

```
allKeys := c.resetKeys[:]
for k := range c.values {
    allKeys = append(allKeys, k)
}
```